src/inc/corprof.idl

```diff
@@ -2408,6 +2412,31 @@ interface ICorProfilerCallback8 : ICorProfilerCallback7
         [in] BOOL       fIsSafeToBlock);
 }
 
+[
+    object,
+    uuid(27583EC3-C8F5-482F-8052-194B8CE4705A),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback9 : ICorProfilerCallback8
+{
+    // This event is triggered whenever a dynamic method is garbage collected
+    // and subsequently unloaded.
+
+    HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
+
+    HRESULT EventPipeEventDelivered(
+        [in] REFGUID eventGuid,
```

Comment:
I suggested that to Brian, but he had a worry that I did not really understand and we ran out of time.   I believe testing was a big part of it.   

My point is that I would like us to AVOID writing MORE code (e.g. tests) that would need to be changed if this changed.   Thus if we are going to use this API, I prefer it to be the right type for the provider (string).  

But it occurs to me that the API surface really needs to be bigger here.   The profiler really needs to be able to send control messages (turn the pipe on or off).     Feels like it is incomplete.